File: Basics-Python/Problem-Set-1/christmas_tree.py
# Here taking the input (tree length) from the user
tree_length = int(input("Give me the tree length you want to display: "))


def print_stars(amount_stars):
    for i in range(amount_stars):
        print("*", end="")


def print_spaces(amount_spaces):
    for i in range(amount_spaces):
        print(" ", end="")


def display(tree_length):
    # initialize the stars with 1
    amount_stars = 1

    for i in range(tree_length):  # for length 5
        print_spaces(tree_length - i - 1)  # 5 - 0 - 1 = 4 spaces
        print_stars(amount_stars)  # prints 1 star
        print()  # return to line (by def it returns)
        amount_stars += 2  # incremente the amount by 2 each iteration


display(tree_length)  # revoke the function

File: Problem_Sets/Problem-Set-1/christmas_tree.c
#include <stdio.h>

void print_stars(int amount_stars) {
    for (int i = 0; i < amount_stars; i++) {
        printf("*");
    }
}

void print_spaces(int amount_spaces) {
    for (int i = 0; i < amount_spaces; i++) {
        printf(" ");
    }
}

void display(int tree_length) {
    int amount_stars = 1;

    for (int i = 0; i < tree_length; i++) {
        print_spaces(tree_length - i - 1);
        print_stars(amount_stars);
        printf("\n");
        amount_stars += 2;
    }
}

int main() {
    int tree_length;
    printf("Give me the tree length you want to display: ");
    scanf("%d", &tree_length);
    display(tree_length);
    return 0;
}